cypress/e2e/CMS/$CancerTypeHomePage/index.js

```diff
@@ -163,5 +140,121 @@ And('user selects {string} from draft {string} dropdown', (dropdown, option) =>
     cy.get(`label:contains("${option}")`).parent().parent().find("select[name*='moderation_state']").select(dropdown)
 })
 
+/*------- Cancer Type Home Page front end verification Starts from below below --------*/
+
+And('the following cards are displayed', (dataTable) => {
+    for (const { title, cardType } of dataTable.hashes()) {
+        cy.get(`div[id='nvcgSlCTHPCards'] h2:contains("${title}")`).should('be.visible')
+    }
+})
+
+And('cthp overview card has description {string}', (title) => {
+    cy.get(`div[class='cardBody'] p:contains("${title}")`).should('be.visible')
+})
+
+And('PDQ link label reads {string}', (title) => {
+    cy.get(`div[class='cardBody'] h4:contains("${title}")`).should('be.visible')
+})
+
+let summaryTitle;
+And('user remembers the title of selected summary for further verification', () => {
+    cy.get('div[id*="pdq-links-current-items"]').then($el => {
+        summaryTitle = $el[0].innerText
+    })
+})
+
+Then('the PDQ link is matching the earlier selected PDQ link', () => {
+    cy.get('div[class*="cthp-treatment"] a').then($el => {
+        const title = $el[0].innerText;
+        expect(summaryTitle).to.include(title);
+    })
+})
 
+let summaryVideo;
+And('user remembers the title of selected video for further verification', () => {
+    cy.get("div[id*='cthp-video-current-']").then($el => {
+        summaryVideo = $el[0].innerText
+    })
+})
+
+Then('the video is matching the earlier selected video', () => {
+    cy.get('div[class*="cthp-screening"] img').then($el => {
+        const video = $el[0].innerText;
+        expect(summaryVideo).to.include(video);
+    })
+})
+
+let summaryPageTitle;
+And('user remembers title of selected Cancer Research List Page for future verification', () => {
+    cy.get("div[class*='cthp-research'] div[class*='label']").then($el => {
+        summaryPageTitle = $el[0].innerText
+    })
+})
+
+Then('the Cancer Research List page title is matching the earlier selected Cancer Research List page title', () => {
+    cy.get("h1").should('include.text', summaryPageTitle)
+
+})
+
+And('user clicks on {string} dropdown', (dropdown) => {
+    cy.get(`div[class='more-info list cthp-dropdown'] label:contains("${dropdown}")`).click({ force: true })
+})
+
+And('the following more info links are displayed', (dataTable) => {
+    for (const { title, url } of dataTable.hashes()) {
+        const replacedTestSiteSection = url.replace("{TEST_SITE_SECTION}", siteSection);
+        cy.get(`ul a[class='title']:contains("${title}")`).should('be.visible').and('have.attr', 'href', replacedTestSiteSection)
+    }
+})
+
+And('cthp causes card has a link {string} with href {string}', (linkText, href) => {
+    const replacedTestSiteSection = href.replace("{TEST_SITE_SECTION}", siteSection);
+    cy.get("div[class='equalheight bgWhite cthp-causes'] a").should('have.text', linkText)
+    cy.get("div[class='equalheight bgWhite cthp-causes'] a").should('have.attr', 'href', replacedTestSiteSection)
```

Comment:
the locator can be changed to div[class*='cthp-causes'] a
also, make use of aliases so you do not repeat your code
cy.get("div[class='equalheight bgWhite cthp-causes'] a").as('card').should('have.text', linkText)
    cy.get('@ card').should('have.attr', 'href', replacedTestSiteSection)